# Importação das bibliotecas

In [196]:
import pandas as pd
import plotly.express as px
import warnings
warnings.simplefilter('ignore')

# Carregamento e Limpeza dos dados

In [197]:
url_gastos = 'https://www.gov.br/secretariageral/pt-br/acesso-a-informacao/informacoes-classificadas-e-desclassificadas/Planilha12003a2022.csv'
url_ipca = 'Arquivos_suporte/IPCA mensal 1994-2022.csv'

In [198]:
gastos_carregados = pd.read_csv(url_gastos,encoding='latin1', delimiter=';')
ipca_carregado = pd.read_csv(url_ipca,encoding='utf-8', delimiter=';')


In [199]:
gastos = gastos_carregados.copy()
ipca = ipca_carregado.copy()

### Análise exploratória

In [200]:
# Identificação inicial das colunas e tipos de dados para o conjunto de dados
# de gastos
gastos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113342 entries, 0 to 113341
Data columns (total 8 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   DATA PGTO               113342 non-null  object 
 1   CPF SERVIDOR            113340 non-null  object 
 2   CPF/CNPJ FORNECEDOR     112906 non-null  float64
 3   NOME FORNECEDOR         113340 non-null  object 
 4   VALOR                   113340 non-null  object 
 5   TIPO                    113340 non-null  object 
 6   SUBELEMENTO DE DESPESA  113340 non-null  object 
 7   CDIC                    113340 non-null  object 
dtypes: float64(1), object(7)
memory usage: 6.9+ MB


Notam-se inicialmente as seguintes informações:

*   113.342 linhas

*   8 colunas

*   Tipo de dados ainda não definidos ou incorretos

*   Dois registros com conteúdo somente na coluna DATA PGTO (possivelmente header ou trailer).

*   A coluna CPF/CNPJ FORNECEDOR possui mais valores nulos do que as demais








In [201]:
# Primeira exibição do dataframe sem tratamentos
display(gastos)

DATA PGTO    CPF SERVIDOR  CPF/CNPJ FORNECEDOR  \
0           02/01/2003  ***.010.008-**         3.134920e+13   
1           02/01/2003  ***.010.008-**         3.134920e+13   
2           02/01/2003  ***.868.251-**         3.134920e+13   
3           03/01/2003  ***.004.131-**         5.927170e+11   
4           03/01/2003  ***.004.131-**         8.393080e+11   
...                ...             ...                  ...   
113337      04/12/2022  ***.136.591-**         4.750841e+13   
113338      04/12/2022  ***.136.591-**         1.255044e+13   
113339      04/12/2022  ***.136.591-**         3.696869e+12   
113340   Fonte: SUPRIM             NaN                  NaN   
113341  At‚ 19/12/2022             NaN                  NaN   

                                          NOME FORNECEDOR          VALOR TIPO  \
0                             JW TRANSPORTADORA TURÖSTICA   R$ 1.696,90     D   
1                             JW TRANSPORTADORA TURÖSTICA   R$ 1.336,60     D   
2                             JW TRANSPORTADORA TURÖSTICA   R$ 1.396,43     D   
3                                         FLORES ALVORADA   R$ 8.585,00     C   
4       CRISTAL LIMPEZA - COMRCIO DE MATERIAIS PARA L...     R$ 251,50     C   
...                                                   ...            ...  ...   
113337               COMPANHIA BRASILEIRA DE DISTRIBUIۂO         518,07    C   
113338                                   RESTAURANTE SOHO         330,52    C   
113339              BIG TRANS COMERCIAL DE ALIMENTOS LTDA          54,66    C   
113340                                                NaN            NaN  NaN   
113341                                                NaN            NaN  NaN   

                              SUBELEMENTO DE DESPESA  \
0                     LOCACAO DE MEIOS DE TRANSPORTE   
1                     LOCACAO DE MEIOS DE TRANSPORTE   
2                     LOCACAO DE MEIOS DE TRANSPORTE   
3       OUTROS SERVI€OS DE TERCEIROS-PESSOA JURÖDICA   
4        MATERIAL DE LIMPEZA E PROD. DE HIGIENIZAۂO   
...                                              ...   
113337                        GENEROS DE ALIMENTAۂO   
113338                   FORNECIMENTO DE ALIMENTAۂO   
113339                        GENEROS DE ALIMENTAۂO   
113340                                           NaN   
113341                                           NaN   

                                                     CDIC  
0       00200.220539/2003-63.R.08.18/12/2002.31/12/2014.N  
1       00200.220539/2003-63.R.08.18/12/2002.31/12/2014.N  
2       00200.600539/2003-63.R.08.18/12/2002.31/12/2014.N  
3       00200.800212/2003-91.R.08.07/12/2002.31/12/2014.N  
4       00200.800212/2003-91.R.08.07/12/2002.31/12/2014.N  
...                                                   ...  
113337  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  
113338  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  
113339  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  
113340                                                NaN  
113341                                                NaN  

[113342 rows x 8 columns]

In [202]:
gastos['TIPO'].unique()

array(['D', 'C', nan], dtype=object)

Após a visualização inicial do DF é possível observar que:
 

*   As duas últimas linhas são irrelevantes para a análise.
*   A coluna "VALOR" possui registro com e sem R$.

*   A coluna "TIPO" refere-se ao tipo de pagamento e possui os valores C e D equivalentes a Crédito e Débito, respectivamente.

In [203]:
# Identificação inicial das colunas e tipos de dados para o conjunto de dados
# de gastos
ipca.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Columns: 342 entries, jul/94 to dez/22
dtypes: float64(341), int64(1)
memory usage: 2.8 KB


In [204]:
display(ipca)

jul/94  ago/94  set/94  out/94  nov/94  dez/94  jan/95  fev/95  março 1995  \
0    6.84    1.86    1.53    2.62    2.81    1.71     1.7    1.02        1.55   

   abr/95  ...  março 2022  abr/22  mai/22  jun/22  jul/22  ago/22  set/22  \
0    2.43  ...        1.62    1.06    0.47    0.67   -0.68   -0.36   -0.29   

   out/22  nov/22  dez/22  
0    0.59    0.41    0.62  

[1 rows x 342 columns]

Nota-se que o df ipca possui uma linha e 342 colunas contendo a variação mensal desde jul/1994 até dez/2022. Além disso, algumas colunas possuem a formatação distinta do padrão das demais.

#### Funções

In [205]:
def presidente(ano: str) -> str:
  """Função aplicada no dataframe de gastos para adição da coluna PRES,
    correspondente ao Presidente eleito no período dos gastos"""
  if ano <= 2010:
      return 'LULA'
  elif ano <= 2016:
      return 'DILMA'
  elif ano <= 2018:
      return "TEMER"
  elif ano <= 2022:
      return "BOLSONARO"

def formata_ano(ano: int) -> int:
  """Função aplicada no dataframe ipca para formataçao do ano"""
  if ano < 23:
      return ano + 2000
  if ano > 23:
      return ano + 1900

### Manipulação do dataframe de gastos

In [206]:
# Elimina as duas últimas linhas
gastos = gastos.iloc[:-2]

# Muda o formato da data para datetime
gastos.loc[:, 'DATA PGTO'] = pd.to_datetime(gastos['DATA PGTO'], errors='coerce', format="%d/%m/%Y")

# Gera uma nova coluna com os anos
gastos.loc[:, 'ANO'] = gastos['DATA PGTO'].dt.year

# Remove os dados faltantes de acordo com a coluna 'CPF SERVIDOR'
gastos = gastos.dropna(axis=0, subset=['CPF SERVIDOR'])

# Cria uma coluna nova com os valores em Reais no formato float
gastos.loc[:, 'VALOR_NUM'] = pd.to_numeric(gastos['VALOR'].str.replace(
    r'R\$', '').str.replace(".", '').str.replace(',', '.').str.strip(), errors='coerce')

# Muda o formato dos dados de tipo(modalidade de pagamento):
gastos.loc[:, "TIPO"] = gastos.loc[:, "TIPO"].replace('D','Débito').replace('C', 'Crédito')

# Remove as dados faltantes que não tenham nenhum valor de gastos
gastos = gastos.dropna(subset=['VALOR_NUM'])

In [207]:
# Gera uma coluna com a informação do nome do atual presidente
gastos['PRES'] = gastos['ANO'].apply(presidente)

### Manipulação do dataframe do ipca

In [208]:
# Realiza a transposição do df e reseta o índice
ipca = ipca.T.reset_index()

# Atribui os nomes das colunas
ipca.columns = ['DATA', 'INFLAÇÃO NO MÊS']

# Trata os registros no campo DATA com o formato indesejado
ipca['DATA'] = ipca['DATA'].str.replace('março ' ,'mar/')
ipca['DATA'] = ipca['DATA'].str.replace('/199' ,'/9')
ipca['DATA'] = ipca['DATA'].str.replace('/200' ,'/0')
ipca['DATA'] = ipca['DATA'].str.replace('/201' ,'/1')
ipca['DATA'] = ipca['DATA'].str.replace('/202' ,'/2')

# Adiciona as colunas temporárias MES e ANO
ipca[['MES','ANO']] = ipca['DATA'].str.split('/',expand=True)

# Adiciona a coluna temporária MES_NUM
mes_para_numero = {'jan':1, 'fev':2, 'mar':3, 'abr':4, 'mai':5, 'jun':6, 
                   'jul':7, 'ago':8, 'set':9, 'out':10, 'nov':11, 'dez':12}
ipca['MES_NUM'] = ipca['MES'].map(mes_para_numero)

# Corrige o formato da coluna ano
ipca['ANO'] = pd.to_numeric(ipca['ANO'], errors='coerce')

# Adiciona a coluna temporária ANO_FORMATADO através da função formata_ano
ipca['ANO_FORMATADO'] = ipca['ANO'].apply(formata_ano)

# Constrói a coluna MES-ANO à partir dos campos MES_NUM e ANO_FORMATADO
ipca['MES-ANO'] = ipca['MES_NUM'].astype(str) + ipca['ANO_FORMATADO'].astype(str)

# Elimina as colunas temporárias
ipca = ipca.drop(['MES', 'ANO', 'MES_NUM', 'ANO_FORMATADO'], axis=1)

In [209]:
# Limita a utilização do df ipca de acordo com as datas existentes no df de gastos
indice_corte = ipca.loc[ipca['MES-ANO'] == '122002'].index[0]
ipca = ipca[indice_corte+1:]

# Adiciona o ponto percentual
ipca['INFLAÇÃO NO MÊS'] = ipca['INFLAÇÃO NO MÊS']/100

# Ordena o df de forma decrescente. Do ano mais recente para o mais antigo.
ipca = ipca.iloc[::-1]

# Adiciona a coluna "AJUSTE INFLAÇÃO" que é o produto acumulado da inflação
# no mês
ipca['AJUSTE INFLAÇÃO'] = (ipca['INFLAÇÃO NO MÊS']+1).cumprod()

### Merge entre os dataframes gastos e ipca

In [210]:
# Constrói a coluna MES-ANO no df de gastos para ser utilizada como pk/fk
gastos.loc[:, 'MES'] = gastos['DATA PGTO'].dt.month
gastos['MES-ANO'] = gastos['MES'].astype(str) + gastos['ANO'].astype(str)

gastos['MES-ANO'] = pd.to_numeric(gastos['MES-ANO'], errors='coerce')
ipca['MES-ANO'] = pd.to_numeric(ipca['MES-ANO'], errors='coerce')

# Merge
gastos = pd.merge(gastos, ipca, how='outer', on='MES-ANO')

In [211]:
gastos['VALOR_CORRIGIDO'] = gastos['VALOR_NUM']*gastos['AJUSTE INFLAÇÃO']
gastos['VALOR_CORRIGIDO'] = gastos['VALOR_CORRIGIDO'].round(2)

In [212]:
display(gastos)

DATA PGTO    CPF SERVIDOR  CPF/CNPJ FORNECEDOR  \
0      2003-01-02  ***.010.008-**         3.134920e+13   
1      2003-01-02  ***.010.008-**         3.134920e+13   
2      2003-01-02  ***.868.251-**         3.134920e+13   
3      2003-01-03  ***.004.131-**         5.927170e+11   
4      2003-01-03  ***.004.131-**         8.393080e+11   
...           ...             ...                  ...   
113319 2022-12-03  ***.136.591-**         1.726166e+13   
113320 2022-12-03  ***.136.591-**         3.987994e+13   
113321 2022-12-04  ***.136.591-**         4.750841e+13   
113322 2022-12-04  ***.136.591-**         1.255044e+13   
113323 2022-12-04  ***.136.591-**         3.696869e+12   

                                          NOME FORNECEDOR          VALOR  \
0                             JW TRANSPORTADORA TURÖSTICA   R$ 1.696,90    
1                             JW TRANSPORTADORA TURÖSTICA   R$ 1.336,60    
2                             JW TRANSPORTADORA TURÖSTICA   R$ 1.396,43    
3                                         FLORES ALVORADA   R$ 8.585,00    
4       CRISTAL LIMPEZA - COMRCIO DE MATERIAIS PARA L...     R$ 251,50    
...                                                   ...            ...   
113319         OUTBACK STEAKHOUSE RESTAURANTES BRASIL S/A          179,5   
113320           BIOMUNDO 105 NORTE COMERCIO DE ALIMENTOS          54,95   
113321               COMPANHIA BRASILEIRA DE DISTRIBUIۂO         518,07   
113322                                   RESTAURANTE SOHO         330,52   
113323              BIG TRANS COMERCIAL DE ALIMENTOS LTDA          54,66   

           TIPO                        SUBELEMENTO DE DESPESA  \
0        Débito                LOCACAO DE MEIOS DE TRANSPORTE   
1        Débito                LOCACAO DE MEIOS DE TRANSPORTE   
2        Débito                LOCACAO DE MEIOS DE TRANSPORTE   
3       Crédito  OUTROS SERVI€OS DE TERCEIROS-PESSOA JURÖDICA   
4       Crédito   MATERIAL DE LIMPEZA E PROD. DE HIGIENIZA€ÇO   
...         ...                                           ...   
113319  Crédito                   FORNECIMENTO DE ALIMENTA€ÇO   
113320  Crédito                        GENEROS DE ALIMENTA€ÇO   
113321  Crédito                        GENEROS DE ALIMENTA€ÇO   
113322  Crédito                   FORNECIMENTO DE ALIMENTA€ÇO   
113323  Crédito                        GENEROS DE ALIMENTA€ÇO   

                                                     CDIC   ANO  VALOR_NUM  \
0       00200.220539/2003-63.R.08.18/12/2002.31/12/2014.N  2003    1696.90   
1       00200.220539/2003-63.R.08.18/12/2002.31/12/2014.N  2003    1336.60   
2       00200.600539/2003-63.R.08.18/12/2002.31/12/2014.N  2003    1396.43   
3       00200.800212/2003-91.R.08.07/12/2002.31/12/2014.N  2003    8585.00   
4       00200.800212/2003-91.R.08.07/12/2002.31/12/2014.N  2003     251.50   
...                                                   ...   ...        ...   
113319  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  2022     179.50   
113320  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  2022      54.95   
113321  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  2022     518.07   
113322  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  2022     330.52   
113323  00264.000642/2022-21.R.08.25/11/2022.31/12/2022.N  2022      54.66   

             PRES  MES  MES-ANO    DATA  INFLAÇÃO NO MÊS  AJUSTE INFLAÇÃO  \
0            LULA    1    12003  jan/03           0.0225         3.173955   
1            LULA    1    12003  jan/03           0.0225         3.173955   
2            LULA    1    12003  jan/03           0.0225         3.173955   
3            LULA    1    12003  jan/03           0.0225         3.173955   
4            LULA    1    12003  jan/03           0.0225         3.173955   
...           ...  ...      ...     ...              ...              ...   
113319  BOLSONARO   12   122022  dez/22           0.0062         1.006200   
113320  BOLSONARO   12   122022  dez/22           0.0062        

## Realização das análises

Neste ponto, é possível realizarmos algumas análises à partir do conjunto de dados tratado. Seguiremos com as seguintes análises:




---


1.   Gastos por presidente acumulado por ano.
2.   Gastos por presidente acumulado por ano corrigidos pelo IPCA.

1.   Gasto acumulado do cartão presidencial por mandato

1.   Gasto acumulado do cartão presidencial por mandato corrigidos pelo IPCA

Abaixo consideramos somente os valores corrigidos pelo IPCA:

5.   Gastos por setores (Subelementos de Despesa)

1.   Despesas com maiores gastos
2.   Análise das despesas no mandato do Lula

- 7.1 - Gastos Lula por modalidade de pagamento
- 7.2 - Gastos Lula por setor

8.   Análise das despesas no mandato do Dilma
- 8.1 - Gastos Dilma por modalidade de pagamento
- 8.2 - Gastos Dilma por setor
9.   Análise das despesas no mandato do Temer
- 9.1 - Gastos Temer por modalidade de pagamento
- 9.2 - Gastos Temer por setor
10.   Análise das despesas no mandato do Bolsonaro
- 10.1 - Gastos Bolsonaro por modalidade de pagamento
- 10.2 - Gastos Bolsonaro por setor


---



#### 1 - Dataframe de gastos por presidente acumulado por ano

In [213]:
gastos_presidente = gastos.groupby(["ANO", "PRES"]).agg({"VALOR_NUM": sum, "VALOR_CORRIGIDO": sum}).reset_index()
gastos_presidente

ANO       PRES   VALOR_NUM  VALOR_CORRIGIDO
0   2003       LULA  5327599.63      15945617.58
1   2004       LULA  6541238.37      18390408.32
2   2005       LULA  5204035.80      13622667.77
3   2006       LULA  4977941.73      12515236.92
4   2007       LULA  3857905.40       9373571.16
5   2008       LULA  6084616.50      13914785.11
6   2009       LULA  5438905.30      11920184.74
7   2010       LULA  6557394.19      13660020.13
8   2011      DILMA  4587842.43       8976242.63
9   2012      DILMA  4598570.59       8542152.62
10  2013      DILMA  6022006.62      10532170.11
11  2014      DILMA  9305096.62      15298335.38
12  2015      DILMA  5706943.28       8635508.04
13  2016      DILMA  4192262.08       5814987.58
14  2017      TEMER  4003637.84       5357091.49
15  2018      TEMER  4866043.60       6293440.89
16  2019  BOLSONARO  5382478.10       6699438.94
17  2020  BOLSONARO  7314318.09       8793959.77
18  2021  BOLSONARO  9927562.29      11081817.57
19  2022  BOLSONARO  4997298.75       5180882.66

In [214]:
fig = px.bar(gastos_presidente, x='ANO', y="VALOR_NUM", color='PRES',
             title="Gastos do cartão presidencial por presidente",
             labels={"VALOR_NUM": "Valor em milhões", 'PRES': "PRESIDENTE"})

fig = fig.update_xaxes(tickangle=45, tickvals=gastos_presidente['ANO'][::])
fig.show()

#### 2 - Dataframe de gastos por presidente acumulado por ano corrigidos pelo IPCA

In [215]:
fig = px.bar(gastos_presidente, x='ANO', y="VALOR_CORRIGIDO", color='PRES',
             title="Gastos do cartão presidencial por presidente corrigidos pelo IPCA",
             labels={"VALOR_CORRIGIDO": "Valor em milhões (corrigido pelo IPCA)", 'PRES': "PRESIDENTE"})

fig = fig.update_xaxes(tickangle=45, tickvals=gastos_presidente['ANO'][::])
fig.show()

#### 3 - Gasto acumulado do cartão presidencial por mandato

In [216]:
lula_mandato_1 = gastos_presidente.loc[0:3].groupby(["PRES"]).agg({"VALOR_NUM": sum}).reset_index()
lula_mandato_1['PRES'] = 'LULA_1'

lula_mandato_2 = gastos_presidente.loc[4:7].groupby(["PRES"]).agg({"VALOR_NUM": sum}).reset_index()
lula_mandato_2['PRES'] = 'LULA_2'

dilma_mandato_1 = gastos_presidente.loc[8:11].groupby(["PRES"]).agg({"VALOR_NUM": sum}).reset_index()
dilma_mandato_1['PRES'] = 'DILMA_1'

dilma_mandato_2 = gastos_presidente.loc[12:13].groupby(["PRES"]).agg({"VALOR_NUM": sum}).reset_index()
dilma_mandato_2['PRES'] = 'DILMA_2'

temer = gastos_presidente.loc[14:15].groupby(["PRES"]).agg({"VALOR_NUM": sum}).reset_index()
bolsonaro = gastos_presidente.loc[16:].groupby(["PRES"]).agg({"VALOR_NUM": sum}).reset_index()
bolsonaro

gastos_acumulados = pd.concat([lula_mandato_1, lula_mandato_2, dilma_mandato_1, dilma_mandato_2, temer, bolsonaro], axis=0, ignore_index=True)
gastos_acumulados

PRES    VALOR_NUM
0     LULA_1  22050815.53
1     LULA_2  21938821.39
2    DILMA_1  24513516.26
3    DILMA_2   9899205.36
4      TEMER   8869681.44
5  BOLSONARO  27621657.23

In [217]:
fig_acumulado = px.bar(gastos_acumulados, x='PRES', y='VALOR_NUM', color='PRES', title='Gasto acumulado do cartão presidencial por mandato',
labels={"VALOR_NUM": "Valor em milhões    *sem correção da inflação", 'PRES': "Presidente"})

fig_acumulado.show()

#### 4 - Gasto acumulado do cartão presidencial por mandato corrigidos pelo IPCA

In [218]:
lula_mandato_1 = gastos_presidente.loc[0:3].groupby(["PRES"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
lula_mandato_1['PRES'] = 'LULA_1'

lula_mandato_2 = gastos_presidente.loc[4:7].groupby(["PRES"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
lula_mandato_2['PRES'] = 'LULA_2'

dilma_mandato_1 = gastos_presidente.loc[8:11].groupby(["PRES"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
dilma_mandato_1['PRES'] = 'DILMA_1'

dilma_mandato_2 = gastos_presidente.loc[12:13].groupby(["PRES"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
dilma_mandato_2['PRES'] = 'DILMA_2'

temer = gastos_presidente.loc[14:15].groupby(["PRES"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
bolsonaro = gastos_presidente.loc[16:].groupby(["PRES"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
bolsonaro

gastos_acumulados_corrigidos = pd.concat([lula_mandato_1, lula_mandato_2, dilma_mandato_1, dilma_mandato_2, temer, bolsonaro], axis=0, ignore_index=True)
gastos_acumulados_corrigidos

PRES  VALOR_CORRIGIDO
0     LULA_1      60473930.59
1     LULA_2      48868561.14
2    DILMA_1      43348900.74
3    DILMA_2      14450495.62
4      TEMER      11650532.38
5  BOLSONARO      31756098.94

In [219]:
fig_acumulado_corrigido = px.bar(gastos_acumulados_corrigidos, x='PRES', y='VALOR_CORRIGIDO', color='PRES', title='Gasto acumulado do cartão presidencial por mandato corrigidos pelo IPCA',
labels={"VALOR_CORRIGIDO": "Valor em milhões *corrigido pelo IPCA", 'PRES': "Presidente"})

fig_acumulado_corrigido.show()

#### 5 - Gastos por setores (Subelementos de Despesa)

In [220]:
gastos_gerais_setores = gastos.groupby(["SUBELEMENTO DE DESPESA"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
gastos_gerais_setores = gastos_gerais_setores.sort_values('VALOR_CORRIGIDO', ascending=False).reset_index()
gastos_gerais_setores

index                      SUBELEMENTO DE DESPESA  VALOR_CORRIGIDO
0       32                                 HOSPEDAGENS      99546541.09
1       35              LOCACAO DE MEIOS DE TRANSPORTE      30888814.08
2       31                      GENEROS DE ALIMENTAۂO      28390010.05
3       28                 FORNECIMENTO DE ALIMENTAۂO      23735146.61
4       87  SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL      12866715.58
..     ...                                         ...              ...
107     74        MATERIAL P/REABILITAۂO PROFISSIONAL           102.71
108     99                   SERVI€OS DE µGUA E ESGOTO            78.06
109     52                          MATERIAL BIOLàGICO            37.80
110     80               OUTRAS DESPESAS COM LOCOMOۂO            23.35
111    102                        SERVI€OS JUDICIµRIOS             6.35

[112 rows x 3 columns]

#### 6 - Despesas com maiores gastos

In [221]:
gastos_gerais_setores.head(10)

index                          SUBELEMENTO DE DESPESA  VALOR_CORRIGIDO
0     32                                     HOSPEDAGENS      99546541.09
1     35                  LOCACAO DE MEIOS DE TRANSPORTE      30888814.08
2     31                          GENEROS DE ALIMENTAۂO      28390010.05
3     28                     FORNECIMENTO DE ALIMENTAۂO      23735146.61
4     87      SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL      12866715.58
5     98                    SERVI€OS DE TELECOMUNICA€åES       2391531.48
6     11              COMBUSTIVEIS E LUBRIF. AUTOMOTIVOS       1983723.14
7     59     MATERIAL DE LIMPEZA E PROD. DE HIGIENIZAۂO       1904220.16
8     36  LOCA€ÇO BENS MOV. OUTR. NATUREZA E INTANGÖVEIS       1607727.91
9     56                      MATERIAL DE COPA E COZINHA        721352.43

#### 7 - Análise das despesas no mandato do Lula

In [222]:
gastos_lula = gastos[gastos['PRES'] == 'LULA']

In [223]:
gastos_lula_setores = gastos_lula.groupby(["TIPO", "SUBELEMENTO DE DESPESA"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
gastos_lula_setores = gastos_lula_setores.sort_values('VALOR_CORRIGIDO', ascending=False).reset_index()
gastos_lula_setores.head(20)

index     TIPO                          SUBELEMENTO DE DESPESA  \
0      18  Crédito                                     HOSPEDAGENS   
1      20  Crédito                  LOCACAO DE MEIOS DE TRANSPORTE   
2      17  Crédito                          GENEROS DE ALIMENTA€ÇO   
3     100   Débito                  LOCACAO DE MEIOS DE TRANSPORTE   
4      14  Crédito                     FORNECIMENTO DE ALIMENTA€ÇO   
5      98   Débito                                     HOSPEDAGENS   
6      94   Débito                     FORNECIMENTO DE ALIMENTA€ÇO   
7      63  Crédito      SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
8     155   Débito                    SERVI€OS DE TELECOMUNICA€åES   
9      40  Crédito     MATERIAL DE LIMPEZA E PROD. DE HIGIENIZA€ÇO   
10     97   Débito                          GENEROS DE ALIMENTA€ÇO   
11      5  Crédito              COMBUSTIVEIS E LUBRIF. AUTOMOTIVOS   
12    147   Débito      SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
13     21  Crédito  LOCA€ÇO BENS MOV. OUTR. NATUREZA E INTANGÖVEIS   
14     37  Crédito                      MATERIAL DE COPA E COZINHA   
15     52  Crédito           MATERIAL P/MANUTEN€ÇO DE BENS IMàVEIS   
16     83   Débito              COMBUSTIVEIS E LUBRIF. AUTOMOTIVOS   
17    142   Débito    OUTROS SERVI€OS DE TERCEIROS-PESSOA JURÖDICA   
18     74  Crédito                 UNIFORMES, TECIDOS E AVIAMENTOS   
19    101   Débito  LOCA€ÇO BENS MOV. OUTR. NATUREZA E INTANGÖVEIS   

    VALOR_CORRIGIDO  
0       44572783.56  
1       21826198.96  
2        9706771.46  
3        8832066.25  
4        6109630.78  
5        4501830.13  
6        2618777.94  
7        1989016.37  
8        1751158.19  
9         817478.86  
10        657999.09  
11        528582.55  
12        457729.37  
13        405034.32  
14        342700.66  
15        318164.88  
16        247922.76  
17        218306.58  
18        202542.39  
19        174335.24

##### 7.1 - Gastos Lula por modalidade de pagamento


In [224]:
fig_lula = px.pie(gastos_lula_setores, values='VALOR_CORRIGIDO', names='TIPO', title='Gastos Lula por modalidade de pagamento',
color_discrete_sequence=px.colors.sequential.RdBu)
fig_lula.show()

##### 7.2 - Gastos de Lula por setor

In [225]:
fig_lula_setores = px.pie(gastos_lula_setores.head(20), values='VALOR_CORRIGIDO', names='SUBELEMENTO DE DESPESA', 
                          title='Principais Gastos de Lula por Setor',color_discrete_sequence=px.colors.sequential.RdBu)
fig_lula_setores.show()

#### 8 - Análise das despesas no mandato da Dilma

In [226]:
gastos_dilma = gastos[gastos['PRES'] == 'DILMA']

gastos_dilma_setores = gastos_dilma.groupby(["TIPO", "SUBELEMENTO DE DESPESA"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
gastos_dilma_setores = gastos_dilma_setores.sort_values('VALOR_CORRIGIDO', ascending=False).reset_index()
gastos_dilma_setores.head(20)

index     TIPO                          SUBELEMENTO DE DESPESA  \
0      21  Crédito                                     HOSPEDAGENS   
1      20  Crédito                          GENEROS DE ALIMENTA€ÇO   
2      17  Crédito                     FORNECIMENTO DE ALIMENTA€ÇO   
3      64  Crédito      SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
4     127   Débito      SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
5      41  Crédito     MATERIAL DE LIMPEZA E PROD. DE HIGIENIZA€ÇO   
6     134   Débito                    SERVI€OS DE TELECOMUNICA€åES   
7      48  Crédito                          MATERIAL FARMACOLOGICO   
8      92   Débito                          GENEROS DE ALIMENTA€ÇO   
9       7  Crédito              COMBUSTIVEIS E LUBRIF. AUTOMOTIVOS   
10     89   Débito                     FORNECIMENTO DE ALIMENTA€ÇO   
11     93   Débito                                     HOSPEDAGENS   
12     97   Débito  LOCA€ÇO BENS MOV. OUTR. NATUREZA E INTANGÖVEIS   
13     38  Crédito                      MATERIAL DE COPA E COZINHA   
14     23  Crédito                  LOCACAO DE MEIOS DE TRANSPORTE   
15     53  Crédito           MATERIAL P/MANUTEN€ÇO DE BENS IMàVEIS   
16     25  Crédito                              LOCA€ÇO DE IMàVEIS   
17     11  Crédito                 DESPESAS COM EXCESSO DE BAGAGEM   
18     22  Crédito                           LIMPEZA E CONSERVA€ÇO   
19     30  Crédito                      MANUT. E CONS. DE VEÖCULOS   

    VALOR_CORRIGIDO  
0       30129151.79  
1        9100242.79  
2        6699945.81  
3        5576936.16  
4         932462.22  
5         708007.78  
6         510051.39  
7         430940.75  
8         414403.51  
9         358124.29  
10        299310.37  
11        245639.67  
12        222323.77  
13        218293.75  
14        182675.71  
15        132456.47  
16        112106.97  
17        112100.74  
18         95214.68  
19         90074.23

##### 8.1 - Gastos de Dilma por modalidade de pagamento

In [227]:
fig_dilma = px.pie(gastos_dilma_setores, values='VALOR_CORRIGIDO', names='TIPO', title='Gastos Dilma por modalidade de pagamento',
color_discrete_sequence=px.colors.sequential.algae)
fig_dilma.show()

##### 8.1 - Gastos de Dilma por setor

In [228]:
fig_dilma_setores = px.pie(gastos_dilma_setores.head(20), values='VALOR_CORRIGIDO', names='SUBELEMENTO DE DESPESA', title='Principais Gastos de Dilma por Setor',
color_discrete_sequence=px.colors.sequential.algae)
fig_dilma_setores.show()

#### 9 - Análise das despesas no mandato do Temer

In [229]:
gastos_temer = gastos[gastos['PRES'] == 'TEMER']

gastos_temer_setores = gastos_temer.groupby(["TIPO", "SUBELEMENTO DE DESPESA"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
gastos_temer_setores = gastos_temer_setores.sort_values('VALOR_CORRIGIDO', ascending=False).reset_index()
gastos_temer_setores.head(20)

index     TIPO                       SUBELEMENTO DE DESPESA  \
0      10  Crédito                                  HOSPEDAGENS   
1       9  Crédito                       GENEROS DE ALIMENTA€ÇO   
2      37  Crédito   SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
3       7  Crédito                  FORNECIMENTO DE ALIMENTA€ÇO   
4      57   Débito   SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
5      23  Crédito  MATERIAL DE LIMPEZA E PROD. DE HIGIENIZA€ÇO   
6       2  Crédito           COMBUSTIVEIS E LUBRIF. AUTOMOTIVOS   
7      50   Débito                  FORNECIMENTO DE ALIMENTA€ÇO   
8      52   Débito                                  HOSPEDAGENS   
9      58   Débito             SERVI€OS DE COMUNICA€ÇO EM GERAL   
10     21  Crédito                   MATERIAL DE COPA E COZINHA   
11     11  Crédito                           LOCA€ÇO DE IMàVEIS   
12     49   Débito                    FESTIVIDADES E HOMENAGENS   
13     38  Crédito                           SERVI€OS BANCµRIOS   
14     18  Crédito     MATERIAL DE ACONDICIONAMENTO E EMBALAGEM   
15     42  Crédito                          SERVI€OS DOMSTICOS   
16      3  Crédito        COMUNICA€ÇO DE DADOS E REDES EM GERAL   
17     60   Débito                 SERVI€OS DE TELECOMUNICA€åES   
18     30  Crédito          MATERIAL P/FESTIVIDADE E HOMENAGENS   
19     45  Crédito                    VARIA€ÇO CAMBIAL NEGATIVA   

    VALOR_CORRIGIDO  
0        4233645.84  
1        3048765.46  
2        1864818.42  
3        1623661.30  
4         256876.82  
5          82666.85  
6          82489.20  
7          72977.47  
8          58971.68  
9          54358.30  
10         54282.95  
11         28834.02  
12         21163.57  
13         17869.11  
14         17363.22  
15         16719.63  
16         14567.24  
17         11627.90  
18          9242.80  
19          9105.83

##### 9.1 - Gastos Temer por modalidade de pagamento

In [230]:
fig_temer = px.pie(gastos_temer_setores, values='VALOR_CORRIGIDO', names='TIPO', title='Gastos Temer por modalidade de pagamento',
color_discrete_sequence=px.colors.sequential.Blues_r)
fig_temer.show()

##### 9.2 - Gastos Temer por setor

In [231]:
fig_temer_setores = px.pie(gastos_temer_setores.head(20), values='VALOR_CORRIGIDO', names='SUBELEMENTO DE DESPESA', title='Principais Gastos de Temer por Setor',
color_discrete_sequence=px.colors.sequential.Blues_r)
fig_temer_setores.show()

#### 10 - Análise das despesas no mandato do Bolsonaro

In [232]:
gastos_bolsonaro = gastos[gastos['PRES'] == 'BOLSONARO']

gastos_bolsonaro_setores = gastos_bolsonaro.groupby(["TIPO", "SUBELEMENTO DE DESPESA"]).agg({"VALOR_CORRIGIDO": sum}).reset_index()
gastos_bolsonaro_setores = gastos_bolsonaro_setores.sort_values('VALOR_CORRIGIDO', ascending=False).reset_index()
gastos_bolsonaro_setores

index     TIPO                      SUBELEMENTO DE DESPESA  \
0      14  Crédito                                 HOSPEDAGENS   
1      11  Crédito                 FORNECIMENTO DE ALIMENTA€ÇO   
2      13  Crédito                      GENEROS DE ALIMENTA€ÇO   
3      73   Débito  SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
4      45  Crédito  SERV.DE APOIO ADMIN.,TECNICO E OPERACIONAL   
..    ...      ...                                         ...   
74     15  Crédito                       LIMPEZA E CONSERVA€ÇO   
75     51  Crédito             SERVI€OS TECNICOS PROFISSIONAIS   
76     67   Débito  MAN. E CONSERV. DE MµQUINAS E EQUIPAMENTOS   
77     38  Crédito                       MATERIAL LABORATORIAL   
78     74   Débito      SERVI€OS DE ESTACIONAMENTO DE VEÖCULOS   

    VALOR_CORRIGIDO  
0       15526743.13  
1        6048891.18  
2        5460734.88  
3         970534.81  
4         818341.41  
..              ...  
74           119.81  
75            87.27  
76            55.09  
77            33.32  
78            17.00  

[79 rows x 4 columns]

##### 10.1 - Gastos Bolsonaro por modalidade de pagamento

In [233]:
fig_bolsonaro = px.pie(gastos_bolsonaro_setores, values='VALOR_CORRIGIDO', names='TIPO', title='Gastos Bolsonaro por modalidade de pagamento',
color_discrete_sequence=px.colors.sequential.YlOrRd)
fig_bolsonaro.show()

##### 10.2 - Gastos Bolsonaro por setor

In [234]:
fig_Bolsonaro_setores = px.pie(gastos_bolsonaro_setores.head(20), values='VALOR_CORRIGIDO', names='SUBELEMENTO DE DESPESA', title='Principais Gastos Bolsonaro por Setor',
color_discrete_sequence=px.colors.sequential.YlOrRd)
fig_Bolsonaro_setores.show()